samples/materialize_workbooks.py

```diff
@@ -60,45 +94,121 @@ def show_materialized_views_status(args, password, site_content_url):
+        return
+    tableau_auth = TSC.TableauAuth(args.username, password, site_content_url)
+    server = TSC.Server(args.server, use_server_version=True)
+    with server.auth.sign_in(tableau_auth):
+        # get all projects with given name
+        projects = [project for project in TSC.Pager(server.projects) if project.name == args.project_name]
+
+        if len(projects) > 1:
+            possible_paths = get_project_paths(server, projects)
+            print("Project name is not unique, use '--project_path <path>'")
+            print("Possible project paths:")
+            print_paths(possible_paths)
+            print('\n')
+            return
+        else:
+            update_project(projects[0], server, enable_materialized_views)
+
+
+def update_project(project, server, enable_materialized_views):
+    for workbook in TSC.Pager(server.workbooks):
+        if workbook.project_id == project.id:
+            workbook.materialized_views_enabled = enable_materialized_views
             server.workbooks.update(workbook)
-            site = server.sites.get_by_content_url(site_content_url)
-            print "Updated materialized views settings for workbook:", workbook.name, "from site:", site.name
-    print
+
+    print("Updated materialized views settings for project: {}".format(project.name))
+    print('\n')
+
+
+def parse_workbook_path(file_path):
+    workbook_paths = open(file_path, 'r')
+    workbook_path_mapping = defaultdict(list)
+    for workbook_path in workbook_paths:
+        workbook_project = workbook_path.rstrip().split('/')
+        workbook_path_mapping[workbook_project[-1]].append('/'.join(workbook_project[:-1]))
+    return workbook_path_mapping
+
+
+def update_workbook(args, enable_materialized_views, password, site_content_url):
+    if args.file_path is None:
+        print("Use '--file-path <file path>' to specify the path of a list of workbooks")
+        print('\n')
+        return
+
+    tableau_auth = TSC.TableauAuth(args.username, password, site_id=site_content_url)
+    server = TSC.Server(args.server, use_server_version=True)
+    with server.auth.sign_in(tableau_auth):
+        workbook_path_mapping = parse_workbook_path(args.file_path)
+        all_projects = {project.id: project for project in TSC.Pager(server.projects)}
+        update_workbooks_by_paths(all_projects, enable_materialized_views, server, workbook_path_mapping)
+
+
+def update_workbooks_by_paths(all_projects, enable_materialized_views, server, workbook_path_mapping):
+    for workbook_name, workbook_paths in workbook_path_mapping.items():
+        req_option = TSC.RequestOptions()
+        req_option.filter.add(TSC.Filter(TSC.RequestOptions.Field.Name,
+                                         TSC.RequestOptions.Operator.Equals,
+                                         workbook_name))
+        workbooks = list(TSC.Pager(server.workbooks, req_option))
+        if len(workbooks) == 1:
```

Comment:
even for only one workbook is there, if the project path is given, we may still want to verify the path matches the workbook name given